h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -133,28 +148,29 @@ int readCatVal(double data, int dataIndex) {
   }
   
   // this method will add to each data row the expanded gam columns
-  public double[] addExpandGamCols(double[] rawData, final RowData rowData) { // add all expanded gam columns here
+  double[] addExpandGamCols(double[] rawData, final RowData rowData) { // add all expanded gam columns here
+    if (!_inputContainGamCols) {
```

Comment:
This `if` confuses me, something is off here `if not _inputContainGamCols` do nothing, `else` add gam cols. Is the name of the variable wrong or am I just not understanding what is happening here?

Reply:
It is a bug.

Reply:
Basically, if the input data have already been gamified, you may not find the gam columns in the input data.  Hence, if that happens, there is no need to add gamified columns to the frame and everything we need to do scoring is already there.